Index: include/datalink.h
===================================================================
--- include/datalink.h	(revision 2858)
+++ include/datalink.h	(working copy)
@@ -89,23 +89,24 @@
 extern "C" {
 #endif /* __cplusplus */
 
-    int datalink_send_pdu(
+    extern bool(*datalink_init) (char *ifname);
+    extern int (*datalink_send_pdu)(
         BACNET_ADDRESS * dest,
         BACNET_NPDU_DATA * npdu_data,
         uint8_t * pdu,
         unsigned pdu_len);
-    extern uint16_t datalink_receive(
+    extern uint16_t (*datalink_receive)(
         BACNET_ADDRESS * src,
         uint8_t * pdu,
         uint16_t max_pdu,
         unsigned timeout);
-    extern void datalink_cleanup(
+    extern void (*datalink_cleanup)(
         void);
-    extern void datalink_get_broadcast_address(
+    extern void (*datalink_get_broadcast_address)(
         BACNET_ADDRESS * dest);
-    extern void datalink_get_my_address(
+    extern void (*datalink_get_my_address)(
         BACNET_ADDRESS * my_address);
-    extern void datalink_set_interface(
+    extern void (*datalink_set_interface)(
         char *ifname);
     extern void datalink_set(
         char *datalink_string);
Index: src/datalink.c
===================================================================
--- src/datalink.c	(revision 2858)
+++ src/datalink.c	(working copy)
@@ -31,6 +31,10 @@
  License.
  -------------------------------------------
 ####COPYRIGHTEND####*/
+#include "ethernet.h"
+#include "bip.h"
+#include "arcnet.h"
+#include "dlmstp.h"
 #include "datalink.h"
 #include <string.h>
 
